src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -328,14 +318,14 @@ def request(self, url, params, headers=None, verb='GET',
         if verbose:
             print(verb, url, params, headers)
         header = self.parse_header(hbuf.getvalue())
+        data = bbuf.getvalue()
+        data = decompress(data, header.header)
```

Comment:
this is not correct since you made it default. Here is what will happen on a client side:
- the server provides response HTTP header with `Content-Encoding: gzip`
- the server data is already decompressed here

As you can see both items contradict each other. You either should remove `Content-Encoding: gzip` header when you decompress by default over here, or you should not decompress here since your server has proper header, i.e. the client should do it.

Reply:
The line `bbuf.getvalue()` simply reads a byte sequence out of the `BytesIO` data object, whatever that data is.
Then in the next line at the `decompress()` call, body data will ONLY be decompressed if the http response header has a `Content-Encoding: gzip`, meaning that the server responded with compressed data and that our client (pycurl_manager, in this case) can decompress the data before passing it over to the upstream modules (note, upstream modules here mean that we are communicating with other services within the WMCore realm).